### PR TITLE
Add boot version to `app info` shell command

### PR DESCRIPTION
**Pull request by onobc**
- Update related tests to Junit5

See #5239

### CODE REVIEW

File: spring-cloud-dataflow-docs/pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	<description>Spring Cloud Data Flow Docs</description>
 	<properties>
 		<main.basedir>${basedir}/..</main.basedir>
-		<docs.resources.version>0.2.1.RELEASE</docs.resources.version>
+		<docs.resources.version>0.2.5</docs.resources.version>
```
Comment (onobc) [inline spring-cloud-dataflow-docs/pom.xml:15]: **[UNRELATED]** `0.2.1.RELEASE` not available in snapshots nor central, `0.2.5` is what we need to be using which is in fact on central.

File: spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java [hunk 2]
```diff
@@ -16,10 +16,10 @@
 
 package org.springframework.cloud.dataflow.shell;
 
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
-import org.junit.Rule;
-import org.junit.rules.TestName;
+import org.junit.jupiter.api.AfterAll;
```
Comment (onobc) [inline spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/AbstractShellIntegrationTest.java:19]: **[UNRELATED]** I converted the app registry test to Junit5 and therefore had to do the same across the test inheritance hieararchy.

File: spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public class ConfigCommandTests {
 
 	@Before
 	public void setUp() {
-		MockitoAnnotations.initMocks(this);
+		MockitoAnnotations.openMocks(this);
```
Comment (onobc) [inline spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/ConfigCommandTests.java:80]: **[UNRELATED]** `initMocks` is deprecated

File: spring-cloud-dataflow-shell/pom.xml [hunk 4]
```diff
@@ -30,6 +30,13 @@
 			<plugin>
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
+				<executions>
+					<execution>
+						<goals>
+							<goal>repackage</goal>
```
Comment (onobc) [inline spring-cloud-dataflow-shell/pom.xml:36]: **[UNRELATED]** this was not being called during `clean install` which resulted in non-executable shell jar.

**Review by corneil — APPROVED** [review]

**Pull request merged**